sources/packages/api/src/services/institution/institution.service.ts

```diff
@@ -227,13 +195,28 @@ export class InstitutionService extends RecordDataModelService<Institution> {
 
     institution.businessGuid = account.institution.guid;
     institution.legalOperatingName = account.institution.legalName;
-
-    await this.createAssociation({
-      institution,
-      user,
-      type: InstitutionUserType.admin,
-      auditUserId: userInfo.userId,
-    });
+    institution.creator = user;
+    // Institution user that has the association between the institution
+    // record and the user record.
+    const institutionUser = new InstitutionUser();
+    institutionUser.creator = user;
+    institutionUser.user = user;
+    institutionUser.institution = institution;
+    // Get admin authorization type.
+    const authorizationType =
+      await this.institutionUserTypeAndRoleRepo.findOneOrFail({
+        type: InstitutionUserType.admin,
+        role: null,
+      });
+    // Assign the new user with an admin authorization.
+    const userAuthorization = new InstitutionUserAuth();
+    userAuthorization.creator = user;
+    userAuthorization.authType = authorizationType;
+    userAuthorization.institutionUser = institutionUser;
+    // Associates the authorizations.
+    institutionUser.authorizations = [userAuthorization];
+    // Saves institution, user, and institution user altogether.
+    await this.institutionUserRepo.save(institutionUser);
 
```

Comment:
Nothing major, but I'm thinking if this should be wrapped (lines 206 to 219) with a transaction manager.

Reply:
Please disregard comment. I was thinking there was another save happening somewhere in between. I will resolve.